dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -1352,25 +1352,43 @@ public void updateReferenceType(String repositoryId, Version version) {
 
     @Override
     protected String getCommitID(String repositoryId, Version version) {
+        try {
+            return getCommitID(repositoryId, version.getReference(), true);
+        } catch (CustomWebApplicationException ex) {
+            return null;
+        }
+    }
+
+    /**
+     * Get the head commit SHA of a specified reference in a particular repository.
+     * @param repositoryId name of the repository (ex 'svonworl/test-notebooks')
+     * @param commitReference reference
+     * @param stripped if true, commitReference does not include the 'refs/{types}/' prefix
+     */
+    public String getCommitID(String repositoryId, String commitReference, boolean stripped) {
         GHRepository repo;
         try {
             repo = github.getRepository(repositoryId);
             GHRef[] refs = getBranchesAndTags(repo);
 
             for (GHRef ref : refs) {
-                String reference = StringUtils.removePattern(ref.getRef(), "refs/.+?/");
-                if (reference.equals(version.getReference())) {
-                    return getCommitSHA(ref, repo, reference);
+                String reference = ref.getRef();
+                String strippedReference = stripReference(reference);
+                if ((stripped ? strippedReference : reference).equals(commitReference)) {
+                    return getCommitSHA(ref, repo, strippedReference);
                 }
-
             }
         } catch (IOException e) {
             LOG.error(gitUsername + ": IOException on getCommitId " + e.getMessage(), e);
-            // this is not so critical to warrant a http error code
+            throw new CustomWebApplicationException("Could not access GitHub reference", HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
What is the reasoning for this being a more severe error now? 
i.e. I would have guessed that the previous version would ignore errors like a missing commit if someone, say, deleted the repo after pushing a change.

Reply:
The diff is confusing, this is actually a new method, coded this way so we can differentiate the error result (thrown exception) from the "ref doesn't exist" result (returns null).

The original `getCommitID` has been adjusted to use this new method, but retain the original behavior (return null in both cases, if the ref doesn't exist, or on error).